File: test-projects/ruby-demo/validation.rb
# frozen_string_literal: true

class Validation
  attr_accessor :name

  def compute(a, b)
    result = a + b
    numbers = [a, b]
    total = numbers.sum
    puts "Total: #{total}"
    result
  end

  def unused
    x = 42
    puts "hello"
  end
end

File: test-projects/go-demo/validation.go
package main

import "fmt"

type Validation struct {
	Name string
}

func (v Validation) Compute(a, b int) int {
	var result string = a + b
	numbers := []int{a, b}
	total := 0
	for _, n := range numbers {
		total += n
	}
	fmt.Printf("Total: %d\n", total)
	return result
}

func (v Validation) Unused() {
	x := 42
	fmt.Println("hello")
}
